ui/src/main/webapp/wizards/ldap/stages/network-settings.js

```diff
@@ -32,87 +43,167 @@ const testConnect = (conn) => ({
+      o[key] = connectionInfo[key]
+      return o
+    }, {})
+
+    onEdit('connectionInfo')(filtered)
+    this.setState({ selected: [] })
+  }
+
+  render () {
+    const {
+      client,
+      setDefaults,
+      prev,
+      next,
+      configs,
+      configs: {
+        connectionInfo = {}
+      } = {},
+      onEdit,
+      onError,
+      onStartSubmit,
+      onEndSubmit
+    } = this.props
+
+    const { messages, ...errors } = groupErrors([
+      'hostname',
+      'port',
+      'encryption'
+    ], this.props.errors)
+
+    const isPortInvalid = configs.port === undefined || configs.port < 0 || configs.port > 65535
+
+    const missingHost = configs.connectionInfo === undefined || Object.keys(configs.connectionInfo).length === 0
+
+    return (
+      <div>
+        <Mount
+          on={setDefaults}
+          port={636}
+          encryption='ldaps'
+          loadbalancing='round_robin' />
+
+        <Title>LDAP Network Settings</Title>
+        <Description>
+          To establish a connection to the remote LDAP store, we need the hostname of the
+          LDAP machine, the port number that the LDAP service is running on, and the
+          encryption method. Typically, port 636 uses LDAPS encryption and port 389 uses
+          StartTLS.  Adding more than one host creates a cluster.  The cluster connections can
+          be load balanced between the hosts (round-robin) or can be treated as a failover cluster
+          where the first host is considered the primary host.
+        </Description>
+
+        <Body>
+          <Hostname
+            value={configs.hostname}
+            errorText={errors.hostname}
+            onEdit={onEdit('hostname')}
+            autoFocus />
+
+          <Port
+            value={configs.port}
+            errorText={isPortInvalid ? getFriendlyMessage('INVALID_PORT_RANGE') : errors.port}
+            onEdit={onEdit('port')}
+            options={[389, 636]} />
+
+          {messages.map((msg, i) => <Message key={i} {...msg} />)}
+
+          <RaisedButton
+            primary
+            style={{margin: '0 auto', marginBottom: '30px', marginTop: '10px', display: 'block'}}
+            label='Add Host'
+            disabled={isPortInvalid}
             onClick={() => {
               onStartSubmit()
               client.query(testConnect({
                 hostname: configs.hostname,
                 port: configs.port,
                 encryption: configs.encryption
-              }))
-                .then(() => {
-                  onEndSubmit()
-                  next('bind-settings')
-                })
-                .catch((err) => {
-                  onEndSubmit()
-                  onError(err)
+              })).then((result) => {
+                onEdit({
+                  hostname: '',
+                  messages: [],
+                  connectionInfo: {
+                    ...connectionInfo,
+                    [Date.now()]: [configs.hostname, configs.port]
```

Comment:
Do we want to use a better data-structure here?

Reply:
Have a suggestion?  I'm open to changing it.  I just modeled it after other areas where a table was used.

Reply:
I think we might want to use an array instead.